contracts/Lock.sol

```diff
@@ -117,25 +123,52 @@ contract Lock is AragonApp, IForwarder, IForwarderFee {
     /**
     * @notice Locks the required amount of tokens and executes the specified action
     * @dev IForwarder interface conformance. Consider using pretransaction on UI for necessary approval.
+    *      We check for edge cases where an approve is made and before this function is called, one of msg.sender current locks is unlocked resulting in a different griefing amount.
     * @param _evmCallScript Script to execute
     */
-    function forward(bytes _evmCallScript) public {
+    function forward(bytes _evmCallScript) public  auth(LOCK_TOKENS_ROLE) {
         require(canForward(msg.sender, _evmCallScript), ERROR_CAN_NOT_FORWARD);
 
+        (uint256 griefAmount, uint256 griefDuration) = getGriefing(msg.sender);
+
+        uint256 allowance = token.allowance(msg.sender, address(this));
+        uint256 computedAmount = lockAmount.add(griefAmount);
+
+        uint256 totalAmount = allowance > computedAmount ? allowance : computedAmount;
```

Comment:
Why are you checking the users allowance and locking the max of allowance and computedAmount? If the user hasn't specified the correct allowance this call will revert. I don't see any need to check the users allowance.

Reply:
It was due to this comment i left at https://github.com/1Hive/lock-app/issues/32#issuecomment-520166230

Not sure how necessary it is to prevent this edge cases.

Reply:
Even if a user has approved more tokens than those required by the lock, intentionally or accidentally, the lock amount deposited should not be changed. Calling approve is quite a generic function and we can't assume the user will use the process provided in the UI. If the user is using the UI and the UI accidentally approves more tokens than needed, when the user next calls approve for this contract it will reset it to exactly the value required. I think it's an edge case the user can risk, especially considering we're not accidentally approving all of the tokens in their account.